redis/client.py

```diff
@@ -1411,11 +1411,30 @@ def get_message(self, ignore_subscribe_messages=False, timeout=0):
         before returning. Timeout should be specified as a floating point
         number.
         """
+        if not self.subscribed and \
+                self.wait_for_subscription(timeout) is False:
+            # The connection isn't subscribed to any channels or patterns, so
+            # no messages are available
+            return None
+
         response = self.parse_response(block=False, timeout=timeout)
```

Comment:
Need to adjust the timeout for the time spent in `wait_for_subscription`.

Reply:
Adjusted on the new commit